src/gennylib/src/context.cpp

```diff
@@ -133,8 +141,10 @@ DefaultRandom& WorkloadContext::getRNGForThread(ActorId id) {
     if (this->isDone()) {
         BOOST_THROW_EXCEPTION(std::logic_error("Cannot create RNGs after setup"));
     }
+
+    std::lock_guard<std::mutex> lk(_rngLock);
     if (auto rng = _rngRegistry.find(id); rng == _rngRegistry.end()) {
-        auto [it, success] = _rngRegistry.try_emplace(id, _rng());
+        auto [it, success] = _rngRegistry.try_emplace(id, _rng + id);
```

Comment:
Ended up using a hash (not visible here since it's treating this as outdated). @dalyd any thoughts on whether that's random enough, or if there's a different approach we should use?